backend/internal/service/handler/reviews/update_review.go

```diff
@@ -0,0 +1,102 @@
+package reviews
+
+import (
+	"github.com/gofiber/fiber/v2"
+	"platnm/internal/models"
+	"strconv"
+	"time"
+
+	"platnm/internal/errs"
+)
+
+type ReviewUpdate struct {
+	ID        int       `json:"id"`
+	UserID    string    `json:"user_id"`
+	Rating    *int      `json:"rating"`
+	Comment   *string   `json:"comment"`
+	UpdatedAt time.Time `json:"updated_at"`
+}
+
+func (h *Handler) UpdateReviewByReviewID(c *fiber.Ctx) error {
+
+	reviewIDStr := c.Params("id")
+
+	// Convert the string reviewID to an integer
+	reviewID, err := strconv.Atoi(reviewIDStr)
+	if err != nil {
+		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{"error": "Invalid review ID"})
+	}
+
+	exists, err := h.reviewRepository.ReviewExists(c.Context(), reviewIDStr)
```

Comment:
something that I think could make life easier in this method is returning the actual review if it exists rather than just a boolean. advantages:
- this review can serve as a base updatedReview, replacing the one you create on line 63. all you'd need to do is update the comment or rating fields if provided
- you can condense your database query to just accept this updatedReview and not the boolean flags, and then have it update both rating and comment in all cases via one query. 
  - what happens if you're only passing in an updated rating and not a comment? since you pulled the existing review and that's your base object, the old comment will already be in updatedReview so you're not losing data. 
    - you'll technically be updating column values to the same value in this case, but I'd argue it's a small price to pay for a shorter and more readable method
  - would eliminate the conditional from your ReviewRepository UpdateReview method
- your conditional in the handler function would be much shorter - you can check if the provided rating/comment are not nil, and then just overwrite the field in the existingReview struct you pulled from the DB
- less code overall!

let me know if this makes sense - sorry that it's a mini-refactor, but I think in the end it's simpler to follow and would also save us future pain if we added a 3rd element to reviews (say a photo or something)

Reply:
Fixed!